doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,408 @@
+At the critical points, the {class}`~.Derivative` equals zero (or is undefined). 
+You can then obtain the function values at those critical points
+by substituting the critical points back into the function using {meth}`~sympy.core.basic.Basic.subs`. You can also tell if the critical point is a maxima or minima by substituting the values
+into the expression for the second derivative: a negative value indicates a maximum, and a positive value indicates a minimum.
+
+```py
+>>> from sympy.abc import x
+>>> from sympy import solve, diff
+>>> f = x**3 + x**2 - x
+>>> derivative = diff(f, x)
+>>> critical_points = solve(derivative, x, dict=True)
+>>> print(critical_points)
+[{x: -1}, {x: 1/3}]
+>>> point1, point2 = critical_points
+>>> print(f.subs(point1))
+1
+>>> print(f.subs(point2))
+-5/27
+>>> curvature = diff(f, x, 2)
+>>> print(curvature.subs(point1))
+>>> print(curvature.subs(point2))
+-4
+4
+```
+
+### {func}`~.solveset` solution sets cannot necessarily be interrogated programmatically
+
+If {func}`~.solveset` returns a finite set (class {class}`~.FiniteSet`), you can iterate through the solutions:
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution_set = solveset(x**2 - y, x)
+>>> print(solution_set)
+{-sqrt(y), sqrt(y)}
+>>> solution_list = list(solution_set)
+>>> print(solution_list)
+[sqrt(y), -sqrt(y)]
+```
+
+However, for more complex results, it may not be possible to list the solutions:
+
+```py
+>>> from sympy import S, solveset, symbols
+>>> x, y = symbols('x, y')
+>>> solution_set = solveset(x**2 - y, x, domain=S.Reals)
+>>> print(solution_set)
+Intersection({-sqrt(y), sqrt(y)}, Reals)
+>>> solution_list = list(solution_set)
+Traceback (most recent call last):
+    ...
+TypeError: The computation had not completed because of the undecidable set membership is found in every candidates.
+```
+
+## Not all equations can be solved
+
+### Equations with no solution
+
+Some equations have no solution, in which case SymPy may return an empty set. For example, the equation 
+$x - 7 = x + 2$ reduces to $-7 = 2$, which has no solution because no value of $x$ will make it true:
+
+```py
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(x - 7, x + 2)
+>>> solve(eqn, x)
+[]
+```
+
+So if SymPy returns an empty list, you may want to check whether there is a mistake in the equation.
+
+### Equations with no analytical solution
+
+The vast majority of arbitrary nonlinear equations are not analytically solvable. 
+The classes of equations that are solvable are basically:
+1. Linear equations
+2. Polynomials, except where limited by the Abel-Ruffini theorem
+3. Equations that can be solved by inverting some transcendental functions
+4. Problems that can be transformed into the cases above 
+(e.g., by turning trigonometric functions into polynomials)
+5. A few other special cases that can be solved with something like the Lambert W function
```

Comment:
This topic probably deserves its own dedicated guide.

Reply:
That's fair. I'm thinking let's leave this in this page for now until this pull request is merged, then I'll split it out into its own page "Which kinds of equations can be solved".